scripts/nrf_provision/fast_pair/fp_provision_cli.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+from intelhex import IntelHex
+import argparse
+import re
+import sys
+import base64
+import math
+
+_FAST_PAIR_MAGIC = (0xFA, 0x57, 0xFA, 0x57)
+_ANTI_SPOOFING_KEY_LEN = 32
+
+
+def align4(address):
+    return 4 * math.ceil(address / 4)
+
+
+def model_id_to_bytes(model_id):
+    if not re.match("^0x[0-9a-fA-F]{6}$", model_id):
+        raise ValueError("Invalid Model ID")
+
+    return bytes.fromhex(model_id.replace("0x", ""))
+
+
+def anti_spoofing_key_to_bytes(anti_spoofing_key):
+    anti_spoofing_key_bytes = base64.b64decode(anti_spoofing_key)
+
+    if len(anti_spoofing_key_bytes) != _ANTI_SPOOFING_KEY_LEN:
+        raise ValueError("Invalid length of Anti Spoofing key")
+
+    return anti_spoofing_key_bytes
+
+
+def prepare_fast_pair_provisioning(out_file, address, model_id, anti_spoofing_key):
+    model_id_bytes = model_id_to_bytes(model_id)
+    anti_spoofing_key_bytes = anti_spoofing_key_to_bytes(anti_spoofing_key)
+
+    out_hex = IntelHex()
+
+    out_hex[address:] = _FAST_PAIR_MAGIC
+    address += align4(len(_FAST_PAIR_MAGIC))
+
+    out_hex[address:] = tuple(model_id_bytes)
+    address += align4(len(model_id_bytes))
+
+    out_hex[address:] = tuple(anti_spoofing_key_bytes)
+    address += align4(len(anti_spoofing_key_bytes))
+
+    out_hex[address:] = _FAST_PAIR_MAGIC
+    address += align4(len(_FAST_PAIR_MAGIC))
```

Comment:
I recommend to add some crc or hash at the end to validate data correctness

Reply:
That's a good idea. Added a Jira task. I think it would be good to already merge existing implementation and then work on it.